File: C++/Baekjoon/Problem1260.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>
using namespace std;

vector<int> graph[1001];
bool visited[1001];

void bfs(int start)
{
    fill_n(visited,1001,false);
    graph->assign(1001,0);
    queue<int> q;
    q.push(start);
    visited[start] = true;

    while(!q.empty())
    {
        int x = q.front();
        q.pop();
        cout<<x<<" ";
        for(int i=0; i<graph[x].size(); i++)
        {
            int y = graph[x][i];
            if(!visited[y])
            {
                q.push(y);
                visited[y] = true;
            }
        }
    }
}

void dfs(int x)
{
    visited[x] = true;
    cout<<x<<" ";
    
    for(int i=0; i<graph[x].size(); i++)
    {
        sort(graph[x].begin(),graph[x].end());
        int y = graph[x][i];
        if(!visited[y])
            dfs(y);
    }
}

int main()
{
    int n,m,s;
    cin>>n>>m>>s;

    for(int i=0; i<m; i++)
    {
        int a,b;
        cin>>a>>b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }
    dfs(s);
    cout<<'\n';
    bfs(s);
}

File: Python/Baekjoon/Problem1260.py
def bfs(start):
    queue = [start]
    visited[start]=True
    while len(queue)>0:
        x = queue[0]
        print(x,end=' ')
        queue.pop(0)

        for i in range(len(graph[x])) :
            y = graph[x][i]
            if not visited[y] :
                queue.append(y)
                visited[y]=True

def dfs(x):
    visited[x] = True
    print(x, end=' ')
    for i in range(len(graph[x])):
        graph[x]=sorted(graph[x])
        y = graph[x][i]
        if not visited[y]: dfs(y)

n, m, v = map(int, input().split())
graph = [[] for _ in range(n + 1)]
for i in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (n + 1)
dfs(v)
print()
visited = [False] * (n + 1)
bfs(v)
